analysis/src/GArepresentativeregion.hpp

```diff
@@ -346,24 +369,350 @@ struct RepresentativeRegion {
-                  << zBounds_Cells[0] << "," << zBounds_Cells[1] << "]" << std::endl;
-        QoIs << "The representative volume specified is bounded by X = [" << xBounds_Meters[0] << ","
-             << xBounds_Meters[1] << "], Y = [" << yBounds_Meters[0] << "," << yBounds_Meters[1] << "], and Z = ["
-             << zBounds_Meters[0] << "," << zBounds_Meters[1] << "] m" << std::endl;
-        QoIs << "The representative volume specified is bounded by cells spanning X = [" << xBounds_Cells[0] << ","
-             << xBounds_Cells[1] << "], Y = [" << yBounds_Cells[0] << "," << yBounds_Cells[1] << "], and Z = ["
-             << zBounds_Cells[0] << "," << zBounds_Cells[1] << "]" << std::endl;
+    // Given the 3D grain structure "GrainID", determine the extent in the direction specified of each of the
+    // "NumberOfGrains" unique grains from the volume bounded by [XLow, XHigh], [YLow, YHigh], [ZLow, ZHigh]. Extent is
+    // calculated in microns
+    void calcGrainExtent(std::vector<float> &GrainExtent, ViewI3D_H GrainID, std::string Direction, double deltax) {
+
+        for (int n = 0; n < NumberOfGrains; n++) {
+            int ThisGrainID = UniqueGrainIDVector[n];
+            int ThisGrainSize = std::round(GrainSizeVector_Microns[n] * convertToCells(deltax, regionType));
+            std::vector<int> GrainCoordinate(ThisGrainSize);
+            int count = 0;
+            for (int k = zBounds_Cells[0]; k <= zBounds_Cells[1]; k++) {
+                for (int i = xBounds_Cells[0]; i <= xBounds_Cells[1]; i++) {
+                    for (int j = yBounds_Cells[0]; j <= yBounds_Cells[1]; j++) {
+                        if (GrainID(k, i, j) == ThisGrainID) {
+                            if (Direction == "X")
+                                GrainCoordinate[count] = i;
+                            else if (Direction == "Y")
+                                GrainCoordinate[count] = j;
+                            else if (Direction == "Z")
+                                GrainCoordinate[count] = k;
+                            count++;
+                        }
+                    }
+                }
+            }
+            int MinCoord = *std::min_element(GrainCoordinate.begin(), GrainCoordinate.end());
+            int MaxCoord = *std::max_element(GrainCoordinate.begin(), GrainCoordinate.end());
+            GrainExtent[n] = (MaxCoord - MinCoord + 1) * convertToMicrons(deltax, "length");
+        }
+    }
+
+    void calcNecessaryGrainExtents(ViewI3D_H GrainID, double deltax) {
+        // If representativeRegion.AnalysisOptions_StatsYN[3] or
+        // representativeRegion.AnalysisOptions_PerGrainStatsYN[5] is toggled, grain extents are needed If
+        // representativeRegion.representativeRegion.AnalysisOptions_StatsYN[4] or
+        // representativeRegion.AnalysisOptions_PerGrainStatsYN[2] is toggled, X extents are needed If
```

Comment:
```suggestion
        // If option 3 or 5 is on, grain extents are needed
        // If option 4 or 2 is on, X extents are needed
```
I should have been more clear, I really just meant that the extra text doesn't help the comment and also causes the issues with auto-formatting

Reply:
Changed somewhere else, but not here